src/assets/assetsUtil.ts

```diff
@@ -34,3 +38,40 @@ export function compareCollectiblesMetadata(
   }, 0);
   return differentValues > 0;
 }
+
+/**
+ * Compares CollectibleContract entries to any CollectibleContract entry.
+ * We need this method when comparing a new fetched CollectibleContract, in case a entry changed to a defined value,
+ * there's a need to update the CollectibleContract object in state.
+ *
+ * @param newCollectibleContract - CollectibleContract data object.
+ * @param oldCollectibleContract - CollectibleContract object to compare with.
+ * @returns Whether there are differences.
+ */
+export function compareCollectibleContract(
+  newCollectibleContract: CollectibleContract,
+  oldCollectibleContract: CollectibleContract,
+) {
+  const keys: (keyof CollectibleContract)[] = [
+    'address',
+    'assetContractType',
+    'createdDate',
+    'description',
+    'externalLink',
+    'logo',
+    'name',
+    'schemaName',
+    'symbol',
+    'totalSupply',
+  ];
+  const differentValues = keys.reduce((value, key) => {
```

Comment:
Maybe it is better we exit as soon as we find a different value using [`keys.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)

Reply:
done here: https://github.com/MetaMask/controllers/pull/680/commits/1be1ed0c058f09d9483d0c93d233657ed117cffd